src/main/java/seedu/address/model/person/Appointment.java

```diff
@@ -2,48 +2,160 @@
 
 import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.format.ResolverStyle;
+import java.time.format.SignStyle;
+import java.time.format.TextStyle;
+import java.time.temporal.ChronoField;
+import java.util.Objects;
+import java.util.function.Function;
+
+import seedu.address.model.person.enums.InputSource;
+import seedu.address.model.person.exceptions.BadAppointmentFormatException;
 
 /**
  * Represents a Person's Appointment in the address book.
  * Guarantees: immutable; is valid as declared in
- * {@link #isValidAppointment(String)}
+ * {@link #isValidAppointmentDelimit(String, InputSource)}
  */
 public class Appointment {
 
-    public static final String MESSAGE_CONSTRAINTS =
-            "Appointment should be in the format YYYY-MM-DD HH:MM HH:MM, and it should not be blank";
+    public static final String MESSAGE_CONSTRAINTS = "Appointment should be in the format "
+            + "\"<Date> <Start Time> <End Time>\" and adhere to the following constraints:\n"
+            + "1. Appointment should not be left blank. Remove this field if no Appointment is needed.\n"
+            + "2. The <Date> field is formatted as \"<Day>-<Month>-<Year>\". It should be separated with dashes (-).\n"
+            + "    ie DD-MM-YYYY.\n"
+            + "    - <Day> is given as a simple numeric value.\n"
+            + "    - <Month> is given as a simple numeric value or a standard 3-letter month name.\n"
+            + "    - <Year> is optional. It is given as a 2 or 4 digit numeric value."
+            + " If no year is given, the current year is assumed.\n"
+            + "3. The 2 <Time> fields are formatted as \"<Hour>:<Minute>\". It should be separated by colons (:) or not"
+            + "separated at all.\n"
+            + "    if no colon is separated, hour should have a leading 0 if relevant. ie HHmm or HH:mm\n"
+            + "4. <Start Time> should be earlier than or equals to <End Time>, and both are defined within <Date>.\n"
+            + "5. <Date>, <Start Time> and <End Time> are separated by spaces ( ) or commas (,).";
```

Comment:
Upon reviewing, it's evident that considerable thought was went into determining the criteria for acceptable appointments. This includes ensuring a valid year (which must not be earlier than the current year), accommodating valid dates (even 30 for February), handling valid months (including the 13th month), validating start and end times, with the end time consistently greater than the start time. I liked this change as opposed to the Appointment as String in which we manipulating the value of the Appoinment was quite limited.

However, I just have a slight concern regarding how verbose the MESSAGE_CONSTRAINTS is.

Reply:
This worry on message constraint verbosity is valid. However, this was done in reference to the `MESSAGE_CONSTRAINTS` of `Email`, which is quite verbose as well.

Your example on 2-digit values is correct - you can refer to `Line 58` for the exact implementation of the formatter. The Builder maps all the 2-digit values to a value in the range of 2000 to 2099.